platform-coop-toolkit/pinecone
# Use SCSS instead of PostCSS with plugins

## Description

After some consideration I feel like it makes sense for us to use SCSS instead of CSS with PostCSS. See:

- https://css-tricks.com/issue-preprocessing-css-custom-properties/
- https://hankchizljaw.com/wrote/keeping-it-simple-with-css-that-scales/#heading-sass-for-the-win!

Notably:

> Basically, what I’m saying is we don’t have to abandon Sass because native language features are coming. It’s probably better to cautiously pre-compile your CSS and not force the browser to work as hard.

Also, I tried writing a bunch of helper classes for applying background colours and writing `.has-<color-name>-background { background: var(--<color-name>); }` over and over again was getting pretty tiresome, as opposed to:

```scss
@each $color in $colors-light {
	.has-#{$color}-background {
		background: var(--#{$color});
		color: var(--black);
	}
}
```

## Steps to test

1. Clone this branch.
2. Install dependencies: `npm i`
3. Lint assets: `npm run lint:styles`
4. Build assets: `npm run production`

**Expected behavior:** Stylesheets lint & build as expected.

## Additional information

Not applicable.

## Related issues

- Resolves #21.
